winforms/demo2/WinForms.Reactive.Client/ViewModels/ItemsViewModel.cs

```diff
@@ -0,0 +1,187 @@
+	[ObservableAsProperty] public IEnumerable<ItemTagDto> SelectedItemTags { get; } 
+	[ObservableAsProperty] public bool IsLoading { get; } 
+	[ObservableAsProperty] public bool HasItems { get; } 
+	[ObservableAsProperty] public bool HasItemSelection { get; } 
+	[ObservableAsProperty] public bool HasSubItems { get; } 
+
+	public ItemsViewModel(
+		IScheduler mainThreadScheduler = null,
+		IItemsService? itemsService = null
+		)
+	{
+		mainThreadScheduler = mainThreadScheduler ?? RxApp.MainThreadScheduler;
+		_itemsService = itemsService ?? Locator.Current.GetService<IItemsService>()!;
+
+		/*
+		 * `LoadItemsCommand` is a command that returns a list from an async call (from an asyc backend, for example).
+		 *
+		 * We guard agains multiple execution via the `canExecute` condition.
+		 * We set a flag `IsExecuting` to show a progressbar/spinner.
+		 * We can handle exceptions if needed.
+		 * We refresh automatically the list every 5 minutes.
+		 */
+		LoadItemsCommand = ReactiveCommand.CreateFromTask(LoadItems, LoadItemsCommand?.IsExecuting.Select(x => !x), mainThreadScheduler);
+		LoadItemsCommand.IsExecuting.ToPropertyEx(this, x => x.IsLoading, initialValue: false);
+		//LoadItemsCommand.ThrownExceptions.Subscribe(error => { /* Handle errors here */ });
+		LoadItemsCommand
+			.ObserveOn(mainThreadScheduler)
+			.ToPropertyEx(this, x => x.Items, initialValue: Enumerable.Empty<ItemDto>());
+
+		var interval = TimeSpan.FromMinutes(5);
+		Observable.Timer(interval, interval, mainThreadScheduler)
+			.Select(time => Unit.Default)
+			.ObserveOn(mainThreadScheduler)
+			.InvokeCommand(this, x => x.LoadItemsCommand);
+
+		ShowDetailsCommand = ReactiveCommand.CreateFromTask(ShowDetails, this.WhenAnyValue(x => x.HasItemSelection), mainThreadScheduler);
+
+		/*
+		 * `WhenAny` is he same as `Observable.CombineLatest`.
+		 * Hence this is equal to
+		 *
+		 * Observable.CombineLatest(
+		 *	this.WhenAnyValue(x => x.Items),
+		 *	this.WhenAnyValue(x => x.SearchStartsWith),
+		 *		this.WhenAnyValue(x => x.SearchEndsWith),
+		 *	(items, startsWith, endsWith) => (items, startsWith, endsWith))
+		 * .Select(...)
+		 * .etc
+		 */
+		this.WhenAny(x => x.Items, x => x.SearchStartsWith, x => x.SearchEndsWith, (items, startsWith, endsWith) => (items, startsWith, endsWith))
+			.Throttle(TimeSpan.FromMilliseconds(800), mainThreadScheduler)
+			.DistinctUntilChanged()
+			.Select(x =>
+			{
+				// uncomment to check how Catch() works
+				//throw new Exception("ahah");
+
+				var r = (x.items.Value ?? Enumerable.Empty<ItemDto>())
+					.Where(i =>
+						(string.IsNullOrEmpty(x.startsWith.Value) || i.Name.StartsWith(x.startsWith.Value))
+						&&
+						(string.IsNullOrEmpty(x.endsWith.Value) || i.Name.EndsWith(x.endsWith.Value)
+						))
+					.ToList();
+				return r;
+			})
+			.Catch(Observable.Return(Enumerable.Empty<ItemDto>()))
+			.ObserveOn(mainThreadScheduler)
+			.ToPropertyEx(this, x => x.ItemsFiltered, initialValue: Enumerable.Empty<ItemDto>());
```

Comment:
it looks like `initialValue` prevents to bind new emitted values, and the ListBox remains empty. If I remove it, it works fine. I have used `initialValue` to be able to set a not-nullable value. If I set a initial value in the property definition (or in the constructor), it's ignored and became `null`. Otherwise I should say that ItemsFiltered is a `IEnumerable<ItemDto>?` (note the question mark), but it looks wrong to me.